api/games/create.js

```diff
@@ -446,35 +446,94 @@ export const createGameFromLobby = gameLobby => {
   const { onRoundStart, onGameStart, onStageStart } = config;
   if ((onGameStart || onRoundStart || onStageStart) && firstRoundId) {
     const game = Games.findOne(gameId);
-    const players = Players.find({
-      _id: { $in: params.playerIds }
-    }).fetch();
-    game.treatment = treatment.factorsObject();
-    game.players = players;
-    game.rounds = Rounds.find({ gameId }).fetch();
-    game.rounds.forEach(round => {
-      round.stages = Stages.find({ roundId: round._id }).fetch();
+    let gameTreatment = null,
+      gamePlayers = null,
+      gameRounds = null;
+
+    Object.defineProperties(game, {
+      treatment: {
+        get() {
+          if (!gameTreatment) {
+            gameTreatment = treatment.factorsObject();
+          }
+
+          return gameTreatment;
+        }
+      },
+      players: {
+        get() {
+          if (!gamePlayers) {
+            gamePlayers = game.getPlayers();
+            gamePlayers.forEach(player => {
+              let round = null,
+                stage = null;
+
+              Object.defineProperties(player, {
+                round: {
+                  get() {
+                    if (!round) {
+                      round = _.extend({}, nextRound);
+                    }
+
+                    return round;
+                  }
+                },
+                stage: {
+                  get() {
+                    if (!stage) {
+                      stage = _.extend({}, nextStage);
+                    }
+
+                    return stage;
+                  }
+                }
+              });
+
+              augmentPlayerStageRound(player, player.stage, player.round, game);
```

Comment:
This does not make sense. You are calling `player.state/round`, just after making them lazy loaded. And in this case, the lazy loading does not serve much purpose, since it's guarding `_.extend()` is not a costly operation (such as querying from the DB).
I think it could be interesting to look at the internals of `augmentPlayerStageRound`, which does queries for  PlayerStages and PlayerRounds, and only add that if people are going to call `get()` on those objects. But that will undoubtably become complicated. Ideally, we'd only load PlayerStages/PlayerRounds if accessed, and group the currently player by player queries into a single query when any of these are accessed, since it's likely if one is accessed, they all will (in a loop). But that's a lot of work, so don't worry about it now, and you can remove the `Object.defineProperties(player...` above.